src/functions/getFirstExistingParentPath.ts

```diff
@@ -8,15 +8,31 @@ import Dependencies from '@/src/types/dependencies'
  * @param dependencies - Dependencies container
  */
 async function getFirstExistingParentPath(directoryPath: string, dependencies: Dependencies): Promise<string> {
-	let parentDirectoryPath = directoryPath
+	let parentDirectoryPath = dependencies.pathNormalize(directoryPath)
 	let parentDirectoryFound = await isDirectoryExisting(parentDirectoryPath, dependencies)
 
-	while (!parentDirectoryFound) {
-		parentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+	const FAILED_TO_FIND_EXISTING_DIRECTORY_VALUE = ''
+
+	/**
+	 * Linux max file path length is 4096 characters.
+	 * With / separators and 1 letter folder names, this gives us a max of ~2048 folders to traverse.
+	 * This is much less error prone than a while loop.
+	 */
```

Comment:
This limit is not valid on every file system
https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits

So... I do not know if I want to add that to this loop.

Reply:
Is there a number we can change it to that would accomplish this goal? Or another way to avoid a while loop/provide a failsafe if we miss another edge case?

Reply:
Ultimately this is of course your call and I'm happy to revert to the while loop if you would like